### DISCUSSION TITLE
Add people to a membership

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#454] [Q&A] [answered]
Using a hook and snippet, I'd like to automatically add a person to a membership when someone cancels their booking. I've looked at the eme_member.php file but I don't know which function to use. Is this possible? I already use the hook 'eme_frontend_cancel_booking_action' but I would like to add some code to add the people to a membership.

**Comment by liedekef** [comment]
There's no simple way of doing that. But, try this for logic:

- create a member using eme_new_member() function (the result will be an array, see at the top of eme_members.php)
- set in that member the person id to the person id you want
- set in that member the membership id to the membership id you want
- get the membership (using eme_get_membership($membership_id))
- insert the member: eme_db_insert_member( $new_member, $membership)
- set paid

So:
```
# new member with person id xx and membership id yy
$new_member = eme_new_member();
$new_member['person_id'] = xx;
$new_member['membership_id'] = yy;

# insert the new member and get the new member id
$membership = eme_get_membership(yy);
$member_id = eme_db_insert_member( $new_member, $membership);

# mark as paid so status gets set etc ...
$member = eme_get_member($member_id);
eme_member_set_paid($member);
```

**Reply by f14m07** [reply]
Thank you for this answer, which I've just tested. Almost everything works fine. The last line of code 
```
eme_member_set_paid($member);
```
doesn't seem to have any effect. The new member is not marked as paid

**Reply by liedekef** [reply]
The only case that can happen is when the member is in fact a family member in a membership. Otherwise you'll need to debug the eme_member_set_paid function ...

**Reply by f14m07** [reply]
Ok, It's a typo on my part. Thanks a lot. 
I still need to work on this hook to initiate the responses associated with this membership.

**Comment by f14m07** [comment]
To solve my problem completely, I now need to insert the answers relating to membership into the table of answers. I know what value I want to insert, but I'm not sure which eme function to use. Can you tell me?

**Reply by liedekef** [reply, marked as answer]
See the function eme_store_member_answers , but it is not easy to add answers via a script currently. EME takes the answers from the form since it needs specific id's etc ...

**Comment by f14m07** [comment]
Maybe, I am going to try to insert directly in the table "answers"
